mdbook-tera-backend/src/main.rs

```diff
@@ -0,0 +1,35 @@
+mod tera_renderer;
+
+use anyhow::anyhow;
+use mdbook::renderer::RenderContext;
+use std::io;
+
+use crate::tera_renderer::custom_component::TeraRendererConfig;
+use crate::tera_renderer::renderer::Renderer;
+
+/// Re-renders HTML files outputed by the HTML backend with Tera templates.
+/// Please make sure the HTML backend is enabled.
+fn main() -> anyhow::Result<()> {
+    let mut stdin = io::stdin();
+    let ctx = RenderContext::from_json(&mut stdin).unwrap();
+    if ctx.config.get_preprocessor("html").is_none() {
+        return Err(anyhow!(
+            "Could not find the HTML backend. Please make sure the HTML backend is enabled."
+        ));
+    }
+    let config: TeraRendererConfig = ctx
+        .config
+        .get_deserialized_opt("output.tera-backend")
+        .expect("Failed to get tera-backend config")
+        .unwrap();
```

Comment:
You can avoid the `unwrap` with an extra `?` (you need to import `anyhow::Context`):

```suggestion
    let config: TeraRendererConfig = ctx
        .config
        .get_deserialized_opt("output.tera-backend")
        .context("Could not load output.tera-backend config")?
        .ok_or(anyhow!("Missing output.tera-backend config"))?;
```